application/controllers/handler.js

```diff
@@ -487,7 +488,10 @@ module.exports = {
       }
 
       const newPassword = require('crypto').randomBytes(9).toString('hex');
-      const hashedPassword = JSSHA.sha512(newPassword + config.SMTP.salt);
+      /* The password is hashed one time at the client site (inner hash and optional) and one time at server-side. As we currently only have one salt, it must be the same for slidewiki-platform and the user-service. In case this is splitted, the user-service must know both salts in order to be able to generate a valid password for resetPassword.*/
+      let hashedPassword = co.hashPassword(newPassword, config.SALT);
```

Comment:
Okay, there's a branch at slidewiki-platform that implements this the other way round. I've overseen that.